packages/client/hmi-client/src/workflow/ops/dataset/tera-dataset-drilldown.vue

```diff
@@ -51,12 +52,16 @@ onMounted(async () => {
 const fetchDataset = async () => {
 	if (!props.node.state?.datasetId) return;
 	const datasetTemp: Dataset | null = await getDataset(props.node.state.datasetId);
-
-	// We are assuming here there is only a single csv file. This may change in the future as the API allows for it.
-	rawContent.value = await downloadRawFile(
-		props.node.state.datasetId,
-		datasetTemp?.fileNames?.[0] ?? ''
-	);
+	if (datasetTemp && datasetTemp.esgfId) {
+		image.value = await getClimateDatasetPreview(datasetTemp.esgfId);
+		rawContent.value = null;
+	} else {
+		// We are assuming here there is only a single csv file. This may change in the future as the API allows for it.
+		rawContent.value = await downloadRawFile(
+			props.node.state.datasetId,
+			datasetTemp?.fileNames?.[0] ?? ''
+		);
+	}
```

Comment:
This would have been better done within the `tera-dataset-description`

Reply:
Should've noticed - I thought that's where the changes happened